### ISSUE TITLE
enhancement to base64 node

### ISSUE THREAD

**Issue by andysc** [node-red/node-red-nodes#401]
could we have an option to make the output a string rather than a buffer, please? 
If you're converting HTTP basic-auth strings, for an example, then you want the resulting string, which means you need an extra "msg.payload = msg.payload.toString();" node to convert it - and you might as well just include the line that does the base64 decode in that same function node, then.

**Comment by dceejay** [comment]
The node should ... if it sees a string input that doesn't match base64 style... try to encode it instead and that should result in a string. Do you have a test case that shows the problem ?

**Comment by andysc** [comment]
OK - so there are actually two problems here.
The one I started with - I'd like a string back, rather than a buffer, after conversion from base64, is:
```
[{"id":"77495546.f174dc","type":"inject","z":"6d47234b.61382c","name":"","topic":"","payload":"YW5keXNj","payloadType":"str","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":140,"y":460,"wires":[["1eb5a6b8.511b69"]]},{"id":"1eb5a6b8.511b69","type":"base64","z":"6d47234b.61382c","name":"","property":"payload","x":320,"y":460,"wires":[["765c8940.490f88","f3b6f78d.31e848"]]},{"id":"765c8940.490f88","type":"function","z":"6d47234b.61382c","name":"stringify","func":"msg.payload = msg.payload.toString();\nreturn msg;","outputs":1,"noerr":0,"x":520,"y":460,"wires":[["60919abe.bdf614"]]},{"id":"f3b6f78d.31e848","type":"debug","z":"6d47234b.61382c","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","x":530,"y":500,"wires":[]},{"id":"60919abe.bdf614","type":"debug","z":"6d47234b.61382c","name":"","active":true,"tosidebar":true,"console":false,"tostatus":true,"complete":"payload","x":710,"y":460,"wires":[]}]
```
having to put a "stringily" function node on the end of the base64 node makes the base64 node rather pointless, as doing the base64 conversion is just one extra line of code in my function node.

But I've found another problem - inconsistent conversion:
```
[{"id":"7c11dd53.2b89cc","type":"inject","z":"6d47234b.61382c","name":"","topic":"","payload":"andy","payloadType":"str","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":130,"y":360,"wires":[["ac7c5c61.c6f99"]]},{"id":"ac7c5c61.c6f99","type":"base64","z":"6d47234b.61382c","name":"","property":"payload","x":300,"y":380,"wires":[["b89c3af4.3b43b8","396b4918.cbe4c6"]]},{"id":"25857c0b.ddd004","type":"inject","z":"6d47234b.61382c","name":"","topic":"","payload":"andysc","payloadType":"str","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":130,"y":400,"wires":[["ac7c5c61.c6f99"]]},{"id":"b89c3af4.3b43b8","type":"debug","z":"6d47234b.61382c","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","x":530,"y":320,"wires":[]},{"id":"396b4918.cbe4c6","type":"base64","z":"6d47234b.61382c","name":"","property":"payload","x":520,"y":380,"wires":[["cd669d4e.2a3808"]]},{"id":"cd669d4e.2a3808","type":"debug","z":"6d47234b.61382c","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","x":710,"y":380,"wires":[]}]
```
'andy' is interpreted as base64 and incorrectly converted to a buffer.
'andysc' is correctly interpreted as a string and converted to base64.

I think we're going to need those "force" options like we've not got in the JSON node (thank you for that!).

**Comment by knolleary** [comment]
The commit that closed this issue has introduced some new issues with the node: see #437 

I can't find anywhere that clearly describes what the node is intended to convert from or to in any of the new modes added and there are no tests for it.

Here's my attempt to list out the modes and their expected behaviours - and associated issues...

 - **Convert between Buffer and Base64** - this is the pre 0.1.0 mode of operation.
    - `Buffer` gets converts to its base64 encoded `String`
    - `String` - if it looks like a Base64 encoded String, it gets converted to a Buffer (in the current version, 0.1.1, this scenario is **broken** as it is returning it as a String not a Buffer - #437 )

 - **Convert String to Base64** - new in 0.1.0
   - A value of *any type* is passed to `Buffer.from()` and converted to a base64 string. So if you pass it what looks like a Base64 string, it will still get converted - you get a base64 encoding of a base64 encoded string. This is *not* the same as the JSON node's "ensure a type" behaviour which leaves a value alone if it is already the right type.
 
- **Convert Base64 to String** - new in 0.1.0
    - `String` - is assumed (without checking) to be a base64 string, passed to `Buffer.from(...,'base64')` and then converted to a String with `.toString('binary');`
    - *Any other type* - causes a runtime error as there's no type checking and it tries to treat it as a String.


### Fixes needed

 1. fix #437 - that is a regression in the node functionality that should not have happened
 2. Rename `Convert String to Base64` to `Encode as Base64` and update the help to indicate any value will be converted to a base64 representation (even if its already a base64 string).
 3. Add type checking to `Convert Base64 to String`, warn & drop if value isn't a String
